src/Build.UnitTests/BinaryLogger_Tests.cs

```diff
@@ -24,67 +26,72 @@ public class BinaryLoggerTests
+        }
 
-                // with this file name, the file will be archived as build artifact so we can inspect it later
-                // this is needed to investigate an intermittent failure of this test on Ubuntu 14
-                var logFilePath = "Microsoft.Build.Engine.UnitTests.dll_TestBinaryLoggerRoundtrip.binlog";
-                binaryLogger.Parameters = logFilePath;
+        [Fact]
+        public void TestBinaryLoggerRoundtrip()
+        {
+            var binaryLogger = new BinaryLogger();
 
-                var mockLogger1 = new MockLogger();
+            binaryLogger.Parameters = _logFile;
 
-                // build and log into binary logger and mockLogger1
-                ObjectModelHelpers.BuildProjectExpectSuccess(s_testProject, binaryLogger, mockLogger1);
+            var mockLogger1 = new MockLogger();
 
-                var mockLogger2 = new MockLogger();
+            // build and log into binary logger and mockLogger1
+            ObjectModelHelpers.BuildProjectExpectSuccess(s_testProject, binaryLogger, mockLogger1);
 
-                var binaryLogReader = new BinaryLogReplayEventSource();
-                mockLogger2.Initialize(binaryLogReader);
+            var mockLogger2 = new MockLogger();
 
-                // read the binary log and replay into mockLogger2
-                binaryLogReader.Replay(logFilePath);
+            var binaryLogReader = new BinaryLogReplayEventSource();
+            mockLogger2.Initialize(binaryLogReader);
 
-                if (File.Exists(logFilePath))
-                {
-                    File.Delete(logFilePath);
-                }
+            // read the binary log and replay into mockLogger2testassembly
+            binaryLogReader.Replay(_logFile);
 
-                Assert.Equal(mockLogger1.FullLog, mockLogger2.FullLog);
-            }
-            finally
-            {
-                Environment.SetEnvironmentVariable(MSBUILDTARGETOUTPUTLOGGING, originalTargetOutputLogging);
-            }
+            Assert.Equal(mockLogger1.FullLog, mockLogger2.FullLog);
         }
 
         [Fact]
         public void NonExistingDirectory()
         {
-            string directory = Path.Combine(ObjectModelHelpers.TempProjectDir, Guid.NewGuid().ToString("N"));
-            string log = Path.Combine(directory, "build.binlog");
+            string directory = Path.GetDirectoryName(_logFile);
             Assert.False(Directory.Exists(directory));
-            Assert.False(File.Exists(log));
-
-            var binaryLogger = new BinaryLogger { Parameters = log };
-
-            try
-            {
-                ObjectModelHelpers.BuildProjectExpectSuccess(s_testProject, binaryLogger);
-                Assert.True(Directory.Exists(directory));
-                Assert.True(File.Exists(log));
-            }
-            finally
-            {
-                ObjectModelHelpers.DeleteDirectory(directory);
-            }
+            Assert.False(File.Exists(_logFile));
```

Comment:
You could use `_env.ExpectFile(".binlog")` which will automatically assert that the file exists at the end (and doesn't create it).